tests/cxx/namespace_use.cc

```diff
@@ -0,0 +1,45 @@
+//===--- namespace_use.cc - test input file for iwyu ----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I .
+
+// Tests that using-directives i.e. `using namespace`, get appropriate
+// include suggestions.
+
+#include "tests/cxx/namespace_use-i1.h"
+
+// Purely using the namespace, not any symbols.
+// This could be unused, or being used transitively in namespace lookup.
+// The include suggestion will be the first header that declares the namespace.
+// IWYU: i1_ns is defined in ...* which isn't directly #included.
```

Comment:
You don't need the ` which isn't directly #included.` tail, the `...*` pattern matches all the way to the end of the line.

Nice touch omitting the filename there, that's a nice way to check that there is a diagnostic even if we can't know for sure for which file.